server/src/middleware/cache.js

```diff
@@ -1,9 +1,9 @@
 const redis_client = require("../config/redis");
 
 module.exports = {
-  checkEventCache(req, res, next) {
-    const { id } = req.params;
-    redis_client.get(`event/${id}`, (err, data) => {
+  checkCache(req, res, next) {
+    const type = req.path.split("/")[2]; // returns blogs, events, etc.
+    redis_client.get(type, (err, data) => {
       if (err) {
         console.log(err);
         res.status(500).json({
```

Comment:
If the key doesn't exist in the cache, then `next()` should be called rather than sending a `500` response.

Reply:
Nit: This should be ideally named getFromCache